protocol-units/bridge/chains/ethereum/src/lib.rs

```diff
@@ -108,14 +108,13 @@ where
 
 	async fn initiate_bridge_transfer(
 		&mut self,
-		// TODO: (richard) we are missing the initiator address here
-		_initiator_address: InitiatorAddress<Self::Address>,
+		initiator_address: InitiatorAddress<Self::Address>,
 		recipient_address: RecipientAddress<Vec<u8>>,
 		hash_lock: HashLock<Self::Hash>,
 		time_lock: TimeLock,
 		amount: Amount,
 	) -> BridgeContractInitiatorResult<()> {
-		let contract = AtomicBridgeInitiator::new(self.initiator_contract.0, &self.rpc_provider);
+		let contract = AtomicBridgeInitiator::new(initiator_address.0 .0, &self.rpc_provider);
```

Comment:
@0xmovses Is the `AtomicBridgeInitiator` not the contract? As such, it should be initialized with the contract address.

Reply:
ah yes, I think the only reason why the tests were passing as it was the same address being used to deploy and then to call. Updating